src/Pate/Discovery.hs

```diff
@@ -405,6 +409,21 @@ archSegmentOffToInterval segOff size =
   let start = PA.addressFromMemAddr (MM.segoffAddr segOff)
   in pure (IM.IntervalCO start (start `PA.addressAddOffset` fromIntegral size))
 
+getBlocks'
+  :: (CMC.MonadThrow m, MS.SymArchConstraints arch, Typeable arch)
```

Comment:
It seems a bit weird to me that we're mixing `MonadThrow` and `MonadError`. Is the `MonadThrow` instance for `EquivM` just lifted from `IO`, and therefore we're throwing an IO exception here rather than giving an error result when this is run from `EquivM`?

Reply:
In `IO` this throws an IO exception. I'm actually not sure what it does in `EquivM`. From my read of the instances in Control.Monad.Catch, that would also throw an IO exception. I'm not thrilled with it either, but there would be some module import loops if we put an `EquivM` in the `HasDedicatedRegister` type